ca_tools/scripts/control_node.py

```diff
@@ -0,0 +1,185 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2013 PAL Robotics SL.
+# Released under the BSD License.
+#
+# Authors:
+#   * Siegfried-A. Gevatter
+
+import curses
+import math
+from enum import Enum
+import signal
+import sys
+
+import rospy
+from geometry_msgs.msg import Twist
+from geometry_msgs.msg import Pose2D
+from geometry_msgs.msg import Point
+from nav_msgs.msg import Odometry
+import tf
+
+TURNING=1
+STOP_TURNING=2
+FORWARD=3
+STOP_FORWARD=4
+
+def signal_handler(signal, frame,queue_size=1):
+  # your code here
+    pubtopic = "/create1/cmd_vel"
+    pub = rospy.Publisher(pubtopic, Twist,queue_size=1)
+    aux = Twist()
+    aux.angular.x=0.0
+    aux.angular.y=0.0
+    aux.angular.z=0.0
+    aux.linear.x=0.0
+    aux.linear.y=0.0
+    aux.linear.z=0.0
+    pub.publish(aux)
+    sys.exit(0)
+
+class ctrl_Node:
+    
+    def __init__(self):
+        
+        rospy.init_node('control_node')
+        pubtopic = "/create1/cmd_vel"
+        self.my_pub = rospy.Publisher(pubtopic, Twist, queue_size=10)
+        subtopic = "/create1/gts"
+        # Create a subscriber with appropriate topic, custom message and name of callback function.
+        self.my_sub=rospy.Subscriber(subtopic,Odometry,self._callback)
+        self.my_goals = []
+        
+        for i in range(4):
+            self.my_goals.append(Pose2D())
+
+
+        self.my_goals[0].x=0.0
+        self.my_goals[0].y=0.0
+
+        self.my_goals[1].x=2.0
+        self.my_goals[1].y=0.0
+
+        self.my_goals[2].x=2.0
+        self.my_goals[2].y=2.0
+
+        self.my_goals[3].x=0.0
+        self.my_goals[3].y=2.0
+        
+        self.goal_num=0
+        self.state=STOP_FORWARD
+        self.my_pose=Pose2D()
+        self.my_pose.x="unset"
+        self.my_pose.y="unset"
+
+# Create a callback function for the subscriber.
+    def _callback(self,data):
```

Comment:
Use [this naming convention](https://www.tutorialsteacher.com/python/private-and-protected-access-modifiers-in-python) for functions and variables in Python. You did it well here but not in the rest of the code.